posts/2020-10-05-Rust-1.47.md

```diff
@@ -25,11 +25,59 @@ appropriate page on our website, and check out the [detailed release notes for
 
 ## What's in 1.47.0 stable
 
-This release contains no new significant features, and is mostly quality of
-life improvements, library stabilizations and const-ifications, and toolchain
+This release contains no new language features, though it does add one
+long-awaited standard library feature. It is mostly quality of life
+improvements, library stabilizations and const-ifications, and toolchain
 improvements. See the [detailed release notes][notes] to learn about other
 changes not covered by this post.
 
+#### Traits on larger arrays
+
+Rust does not currently have a way to be generic over integer values. This
+has long caused problems with arrays, because arrays have an integer as part
+of their type; `[T; N]` is the type of an array of type `T` of `N` length.
+Because there is no way to be generic over `N`, you have to manually implement
+traits for arrays for every `N` you want to support. For the standard library,
+it was decided to support up to `N` of 32.
+
+We have been working on a feature called "const generics" that would allow
+you to be generic over `N`. Fully explaining this feature is out of the scope
+of this post, because we are not stabilizing const generics just yet.
+However, the core of this feature has been implemented in the compiler, and
+it has been decided that the feature is far enough along that we are okay
+with [the standard library using it to implement traits on arrays of any
+length](https://github.com/rust-lang/rust/pull/74060/). What this means in
+practice is that if you try to do something like this on Rust 1.46:
+
+```rust
+fn main() {
+    let xs = [0; 34];
+    
+    println!("{:?}", xs);
+}
+```
+
+you'd get this error:
+
+```text
+error[E0277]: arrays only have std trait implementations for lengths 0..=32
+ --> src/main.rs:4:22
+  |
+4 |     println!("{:?}", xs);
+  |                      ^^ the trait `std::array::LengthAtMost32` is not implemented for `[{integer}; 34]`
+  |
+  = note: required because of the requirements on the impl of `std::fmt::Debug` for `[{integer}; 34]`
+  = note: required by `std::fmt::Debug::fmt`
+  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
+```
+
+But with Rust 1.47, it will properly print out.
```

Comment:
Not sure how much this adds, but
```suggestion
But with Rust 1.47, it will properly print out `[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`.
```

Reply:
Yeah, I thought maybe this was a bit too spammy, so I left it out. If people prefer it in, I'm okay with that.

Reply:
I don't think we should say what it prints out, but it currently feels a bit cut off -- perhaps "it will properly print out the array"?